packages/core/src/odata/v2/request-builder/response-data-accessor.ts

```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. */
```

Comment:
General: Perhaps it makes sense to anticipate the fact that C4C,SFSF, S/4 all not obey the OData specs to 100% and include some checks on the types of the results i.e. include a warn/error statement in the  getSingle and  getCollection in case the data type is not object or array. This would make potential issue opene by consumer easier to classify.

Reply:
True. Would you also add that for the existing workaround? (My opinion: yes).